6211 Карпинская Анастасия Михайловна Лаб.3 Вар.8

stasya-kk opened the pull request:
ФИО: Карпинская Анастасия
Номер группы: 6211  
Номер лабораторной: 3
Номер варианта: 8  
Краткое описание решаемой задачи: Согласно варианту 8 лабораторной 3 мне надо было поменять местами каналы изображения.  
Краткое описание работы кода: Модуль image_processor.py загружает изображения и меняет местами цветовые каналы (R→G, G→B, B→R). Модуль visualization.py отображает оригинальное и обработанное изображения с помощью matplotlib.
Модуль main.py является точкой входа в приложение. для запуска программы необходимо указать python main.py --input "images/000001.jpg" --output "images/result.jpg" ( сначала путь к изображению, потом путь для сохранения)

Commit:
сделала лабу 1

Commit:
Исправила ошибки, также теперь для запуска теперь нужно вводить python main.py data.txt moscow_people.txt

stasya-kk committed:
Merge branch 'itsecd:main' into main

stasya-kk committed:
сделана лаба 3

stasya-kk committed:
Delete moscow_people.txt

microalan reviewed (commented):
1. Добавить type-hints
2. Дать возможность пользователю выбирать новый порядок каналов

stasya-kk committed:
внесла изменения

теперь для запуска программы необходимо вводить  python main.py --input "images/000001.jpg" --output "images/result.jpg" --channels 1,2,0 (каналы можно изменять)

microalan reviewed (approved):
Зачтено
